core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java

```diff
@@ -0,0 +1,327 @@
+
+                            path.trimTo(snapshotLen).concat(configuration.getDbDirectory()).concat(tableName).slash$();
+                            if (ff.mkdirs(path, configuration.getMkDirMode()) != 0) {
+                                throw CairoException.instance(ff.errno()).put("Could not create [dir=").put(path).put(']');
+                            }
+
+                            int rootLen = path.length();
+                            // Copy _meta file.
+                            path.trimTo(rootLen).concat(TableUtils.META_FILE_NAME).$();
+                            mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                            reader.getMetadata().dumpTo(mem);
+                            mem.close(false);
+                            // Copy _txn file.
+                            path.trimTo(rootLen).concat(TableUtils.TXN_FILE_NAME).$();
+                            mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                            reader.getTxFile().dumpTo(mem);
+                            mem.close(false);
+                            // Copy _cv file.
+                            path.trimTo(rootLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
+                            mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                            reader.getColumnVersionReader().dumpTo(mem);
+                            mem.close(false);
+
+                            LOG.info().$("snapshot copied [table=").$(tableName).$(']').$();
+                        }
+
+                        // Write instance id to the snapshot metadata file.
+                        path.trimTo(snapshotLen).concat(TableUtils.SNAPSHOT_META_FILE_NAME).$();
+                        mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                        mem.putStr(configuration.getSnapshotInstanceId());
+                        mem.close();
+
+                        // Flush dirty pages and filesystem metadata to disk
+                        if (ff.sync() != 0) {
+                            throw CairoException.instance(ff.errno()).put("Could not sync");
+                        }
+
+                        LOG.info().$("snapshot copying finished").$();
+                    } catch (Throwable e) {
+                        unsafeReleaseReaders();
+                        LOG.error()
+                                .$("snapshot prepare error [e=").$(e)
+                                .I$();
+                        throw e;
+                    }
+                }
+            }
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public void completeSnapshot() throws SqlException {
+        if (!lock.tryLock()) {
+            throw SqlException.position(0).put("Another snapshot command in progress");
+        }
+        try {
+            if (snapshotReaders.size() == 0) {
+                throw SqlException.position(0).put("SNAPSHOT PREPARE must be called before SNAPSHOT COMPLETE");
```

Comment:
```suggestion
                LOG.info().$("Snapshot it empty, SNAPSHOT COMPLETE is ignored.").$();
                return;
```

Reply:
Changed in 9ffaccfd3d83c6cc4d3cc94c2152edb674726a47